test/e2e/tests/confirmations/transactions/erc1155-set-approval-for-all-redesign.spec.ts

```diff
@@ -0,0 +1,125 @@
+
+const {
+  defaultGanacheOptions,
+  defaultGanacheOptionsForType2Transactions,
+  withFixtures,
+} = require('../../../helpers');
+const FixtureBuilder = require('../../../fixture-builder');
+const { SMART_CONTRACTS } = require('../../../seeder/smart-contracts');
+
+describe('Confirmation Redesign ERC1155 setApprovalForAll', function () {
+  const smartContract = SMART_CONTRACTS.NFTS;
+
+  describe('Submit an revoke transaction @no-mmi', function () {
+    it('Sends a type 0 transaction (Legacy)', async function () {
+      await withFixtures(
+        {
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withPermissionControllerConnectedToTestDapp()
+            .withPreferencesController({
+              preferences: {
+                redesignedConfirmationsEnabled: true,
+                isRedesignedConfirmationsDeveloperEnabled: true,
+              },
+            })
+            .build(),
+          ganacheOptions: defaultGanacheOptions,
+          smartContract,
+          testSpecificMock: mocks,
+          title: this.test?.fullTitle(),
+        },
+        async ({ driver, contractRegistry }: TestSuiteArguments) => {
+          await createTransactionAndAssertDetails(driver, contractRegistry);
+        },
+      );
+    });
+
+    it('Sends a type 2 transaction (EIP1559)', async function () {
+      await withFixtures(
+        {
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withPermissionControllerConnectedToTestDapp()
+            .withPreferencesController({
+              preferences: {
+                redesignedConfirmationsEnabled: true,
+                isRedesignedConfirmationsDeveloperEnabled: true,
+              },
+            })
+            .build(),
+          ganacheOptions: defaultGanacheOptionsForType2Transactions,
+          smartContract,
+          testSpecificMock: mocks,
+          title: this.test?.fullTitle(),
+        },
+        async ({ driver, contractRegistry }: TestSuiteArguments) => {
+          await createTransactionAndAssertDetails(driver, contractRegistry);
+        },
+      );
+    });
+  });
+});
+
+async function mocks(server: MockttpServer) {
+  return [await mocked4BytesSetApprovalForAll(server)];
+}
+
+export async function mocked4BytesSetApprovalForAll(mockServer: MockttpServer) {
+  return await mockServer
+    .forGet('https://www.4byte.directory/api/v1/signatures/')
+    .withQuery({ hex_signature: '0xa22cb465' })
+    .always()
+    .thenCallback(() => ({
+      statusCode: 200,
+      json: {
+        count: 1,
+        next: null,
+        previous: null,
+        results: [
+          {
+            bytes_signature: '¢,´e',
+            created_at: '2018-04-11T21:47:39.980645Z',
+            hex_signature: '0xa22cb465',
+            id: 29659,
+            text_signature: 'setApprovalForAll(address,bool)',
+          },
+        ],
+      },
+    }));
+}
+
+async function createTransactionAndAssertDetails(
```

Comment:
Rather than creating additional test specific helpers, has the moment come that we can create a `TransactionConfirmation` page object model? 😃 

We don't have to refactor old tests, but seems a worthy use of time if we're adding additional tests?

I really believe it will make a staggering difference to test duplication, readability, and development cost.